libs/gl-client/src/lnurl/pay/mod.rs

```diff
@@ -123,6 +173,47 @@ fn ensure_amount_is_within_range(
     Ok(())
 }
 
+//LUD-16: Paying to static internet identifiers.
+pub fn parse_lightning_address(lightning_address: &str) -> Result<String> {
+    let lightning_address_components: Vec<&str> = lightning_address.split("@").collect();
+
+    let username = lightning_address_components.get(0);
+
+    if username.is_none() {
+        return Err(anyhow!("Could not parse username in lightning address"));
+    }
```

Comment:
This should likely be `is_empty()` because `@mydomain.com` will produce a valid split, but the username is empty. Also without the 2 part check above, this will never get triggered:

```
mydomain.com -> ["mydomain.com"] -> valid username, but empty domain?
```